ipnsfs/system.go

```diff
@@ -0,0 +1,291 @@
+package ipnsfs
```

Comment:
this package needs a doc explaining its design. i dont readily understand what's going on, why a keyroot is needed, what the semantics are, etc.